internal/webhook/sparkpod_defaulter.go

```diff
@@ -191,7 +191,7 @@ func addVolume(pod *corev1.Pod, volume corev1.Volume) error {
 func addVolumeMount(pod *corev1.Pod, mount corev1.VolumeMount) error {
 	i := findContainer(pod)
 	if i < 0 {
-		logger.Info("not able to add VolumeMount %s as Spark container was not found in pod %s", mount.Name, pod.Name)
+		return fmt.Errorf("failed to add volumeMounts as Spark container not found")
 	}
```

Comment:
why not modify findContainer to return a tuple (int, error) instead ?
And call it earlier in the code instead of every patch  operation

Reply:
Yes, the logic of findContainer can be optimized as you said, maybe we can do it in a separated PR.